launchcontainer/static/js/src/launchcontainer.js

```diff
@@ -1,33 +1,74 @@
+  link.setAttribute('href', path);
 
-    port = (link.port) ? ':'+link.port : '';
-    return link.protocol + '//' + link.hostname + port;
+  port = (link.port) ? ':'+link.port : '';
+  return link.protocol + '//' + link.hostname + port;
 }
 
 function LaunchContainerXBlock(runtime, element) {
 
-    $(document).ready(
-      function () {
-        var $launcher = $('#launcher1'), $launch_button = $launcher.find('input');
-        $launch_button.click(function() {
-            console.log('Clicked launch button');
-            $launch_button.attr('disabled', 'disabled').val('Launching...');
-            $launcher.find('iframe')[0].contentWindow.postMessage({
-                owner_email: "{{ user_email }}",
-                project: "{{ project }}"
-            }, "{{ API_url }}");
-            return false;
-        });
-        window.addEventListener("message", function (event) {
-            if (event.origin !== getURLOrigin('{{ API_url }}')) return;
-            if(event.data.status === 'siteDeployed') {
-                $launcher.html(event.data.html_content);
-            } else if(event.data.status === 'deploymentError') {
-                $launcher.text(event.data.error_message);
+  $(document).ready(
+    function () {
+      // Submit the data to the AVL server and process the response. //
+      var $launcher = $('#launcher1'),
+          $post_url = '{{ API_url }}',
+          $launcher_form = $('#launcher_form'),
+          $launcher_submit = $('#launcher_submit'),
+          $launcher_email = $('#launcher_email'),
+          $launch_notification = $('#launcher_notification'), 
+          $waiting = 'Your request for a lab is being processed, and may take up to 90 seconds to start. '
+          + 'If you are having issues, please <a href="mailto:technical@appsembler.com">notify us</a>.'
+
+      // This is for studio: If there is no email addy, 
+      // you can enter it manually.
+      if (!$launcher_email.val()) {
+        $launcher_email.removeClass('hide');
+      }
+
+      $('#launcher_form').submit(function (event) {
+
+        // Shut down the buttons.
+        event.preventDefault();
+        $launcher_submit.disabled = true; 
+        $launcher_submit.val('Launching ...');
+        $launch_notification.html($waiting);
+        $launch_notification.removeClass('hide')
+                            .removeClass('ui-state-error')
+                            .removeClass('ui-state-notification');
+        $launcher.find('iframe')[0].contentWindow.postMessage($(this).serialize(), "{{ API_url }}");
+        return false;
+      });
+
+      window.addEventListener("message", function (event) {
+        if (event.origin !== getURLOrigin('{{ API_url }}')) return;
+        if(event.data.status === 'siteDeployed') {
+          $launch_notification.removeClass('hide')
+                              .removeClass('ui-state-error')
+                              .html(event.data.html_content);
```

Comment:
We are returning an `html` snippet (for now). If I change this to `.text()`, will it not escape the elements instead of rendering the html? https://stackoverflow.com/a/14504818/785400